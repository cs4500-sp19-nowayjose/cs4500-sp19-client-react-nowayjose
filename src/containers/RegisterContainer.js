import React from 'react'
import UserService from '../services/UserService'
import Register from '../components/Register/Register'

export default class RegisterContainer extends React.Component {
    constructor(props) {
        super(props);

        this.userService = UserService.getInstance()

        this.state = {
            firstName: "",
            lastName: "",
            username: "",
            password: ""
        };
    }

    handleChange = event => {
        this.setState({
            [event.target.id]: event.target.value
        });
    }

    handleCreate = event => {
        var user = this.state; 
        this.userService.registerUser(user).then(resp => {
            console.log(resp)
        })
    }

    render() {
        return (<Register
            handleChange={this.handleChange}
            handleCreate={this.handleCreate}
        />);
    };
}