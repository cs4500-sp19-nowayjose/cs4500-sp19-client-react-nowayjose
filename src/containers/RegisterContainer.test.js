import React from 'react'
import RegisterContainer from './RegisterContainer'
import TestRenderer from 'react-test-renderer';

test('[RegisterContainer renders correctly]', () => {
    const testRenderer = TestRenderer.create(
        <RegisterContainer/>)
    let tree = testRenderer.toJSON()
    expect(tree).toMatchSnapshot()

    tree = testRenderer.toJSON()
    expect(tree).toMatchSnapshot()
})

test('[RegisterContainer renders correctly after hitting create]', () => {
    const testRenderer = TestRenderer.create(
        <RegisterContainer/>)
    let tree = testRenderer.toJSON()
    expect(tree).toMatchSnapshot()

    const testInstance   = testRenderer.root
    let   createButton    = testInstance.findByProps({id: 'create'})
    createButton.props.onClick({})

    tree = testRenderer.toJSON()
    expect(tree).toMatchSnapshot()
})