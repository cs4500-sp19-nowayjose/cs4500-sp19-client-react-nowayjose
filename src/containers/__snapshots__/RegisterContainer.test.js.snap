// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[RegisterContainer renders correctly after hitting create] 1`] = `
<div>
  <h1>
    Create your account
  </h1>
  <br />
  <div>
    <div
      className="row"
    >
      <div
        className="col-6"
      >
        <label
          for="first-name"
        >
          First name
        </label>
        <input
          className="first-name form-control"
          id="firstName"
          onChange={[Function]}
        />
      </div>
      <div
        className="col-6"
      >
        <label
          for="last-name"
        >
          Last name
        </label>
        <input
          className="last-name form-control"
          id="lastName"
          onChange={[Function]}
        />
      </div>
    </div>
    <br />
    <div
      className="row"
    >
      <div
        className="col-12"
      >
        <label
          for="username"
        >
          Username/Email
        </label>
        <input
          className="email form-control"
          id="username"
          onChange={[Function]}
        />
      </div>
    </div>
    <br />
    <div
      className="row"
    >
      <div
        className="col-12"
      >
        <label
          for="password"
        >
          Password
        </label>
        <input
          className="password form-control"
          id="password"
          onChange={[Function]}
        />
      </div>
    </div>
    <br />
    <div
      className="row"
    >
      <div
        className="col-12"
      >
        <button
          className="btn btn-primary btn-block"
          id="create"
          onClick={[Function]}
        >
          Create Account
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`[RegisterContainer renders correctly after hitting create] 2`] = `
<div>
  <h1>
    Create your account
  </h1>
  <br />
  <div>
    <div
      className="row"
    >
      <div
        className="col-6"
      >
        <label
          for="first-name"
        >
          First name
        </label>
        <input
          className="first-name form-control"
          id="firstName"
          onChange={[Function]}
        />
      </div>
      <div
        className="col-6"
      >
        <label
          for="last-name"
        >
          Last name
        </label>
        <input
          className="last-name form-control"
          id="lastName"
          onChange={[Function]}
        />
      </div>
    </div>
    <br />
    <div
      className="row"
    >
      <div
        className="col-12"
      >
        <label
          for="username"
        >
          Username/Email
        </label>
        <input
          className="email form-control"
          id="username"
          onChange={[Function]}
        />
      </div>
    </div>
    <br />
    <div
      className="row"
    >
      <div
        className="col-12"
      >
        <label
          for="password"
        >
          Password
        </label>
        <input
          className="password form-control"
          id="password"
          onChange={[Function]}
        />
      </div>
    </div>
    <br />
    <div
      className="row"
    >
      <div
        className="col-12"
      >
        <button
          className="btn btn-primary btn-block"
          id="create"
          onClick={[Function]}
        >
          Create Account
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`[RegisterContainer renders correctly] 1`] = `
<div>
  <h1>
    Create your account
  </h1>
  <br />
  <div>
    <div
      className="row"
    >
      <div
        className="col-6"
      >
        <label
          for="first-name"
        >
          First name
        </label>
        <input
          className="first-name form-control"
          id="firstName"
          onChange={[Function]}
        />
      </div>
      <div
        className="col-6"
      >
        <label
          for="last-name"
        >
          Last name
        </label>
        <input
          className="last-name form-control"
          id="lastName"
          onChange={[Function]}
        />
      </div>
    </div>
    <br />
    <div
      className="row"
    >
      <div
        className="col-12"
      >
        <label
          for="username"
        >
          Username/Email
        </label>
        <input
          className="email form-control"
          id="username"
          onChange={[Function]}
        />
      </div>
    </div>
    <br />
    <div
      className="row"
    >
      <div
        className="col-12"
      >
        <label
          for="password"
        >
          Password
        </label>
        <input
          className="password form-control"
          id="password"
          onChange={[Function]}
        />
      </div>
    </div>
    <br />
    <div
      className="row"
    >
      <div
        className="col-12"
      >
        <button
          className="btn btn-primary btn-block"
          id="create"
          onClick={[Function]}
        >
          Create Account
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`[RegisterContainer renders correctly] 2`] = `
<div>
  <h1>
    Create your account
  </h1>
  <br />
  <div>
    <div
      className="row"
    >
      <div
        className="col-6"
      >
        <label
          for="first-name"
        >
          First name
        </label>
        <input
          className="first-name form-control"
          id="firstName"
          onChange={[Function]}
        />
      </div>
      <div
        className="col-6"
      >
        <label
          for="last-name"
        >
          Last name
        </label>
        <input
          className="last-name form-control"
          id="lastName"
          onChange={[Function]}
        />
      </div>
    </div>
    <br />
    <div
      className="row"
    >
      <div
        className="col-12"
      >
        <label
          for="username"
        >
          Username/Email
        </label>
        <input
          className="email form-control"
          id="username"
          onChange={[Function]}
        />
      </div>
    </div>
    <br />
    <div
      className="row"
    >
      <div
        className="col-12"
      >
        <label
          for="password"
        >
          Password
        </label>
        <input
          className="password form-control"
          id="password"
          onChange={[Function]}
        />
      </div>
    </div>
    <br />
    <div
      className="row"
    >
      <div
        className="col-12"
      >
        <button
          className="btn btn-primary btn-block"
          id="create"
          onClick={[Function]}
        >
          Create Account
        </button>
      </div>
    </div>
  </div>
</div>
`;
